infrastructure/builder/src/generate/generators/k8s.js

```diff
@@ -0,0 +1,354 @@
+          for (const path of ingress.paths) {
+            ingresses.push({
+              path,
+              projectName: ingress.projectName,
+            });
+          }
+        }
+      }
+    }
+    const templates = requirements['k8s-templates'];
+    const rendered = jsone(templates['ingress'], {ingresses});
+    await writeRepoYAML(path.join(TMPL_DIR, 'ingress.yaml'), rendered);
+  },
+});
+
+exports.tasks.push({
+  title: `Generate values.yaml and values.schema.yaml`,
+  requires: [...SERVICES.map(name => `configs-${name}`), ...SERVICES.map(name => `procslist-${name}`)],
+  provides: [],
+  run: async (requirements, utils) => {
+    const schema = {
+      '$schema': 'http://json-schema.org/draft-06/schema#',
+      type: 'object',
+      title: 'Taskcluster Configuration Values',
+      properties: {
+        rootUrl: {
+          type: 'string',
+          format: 'uri',
+          description: 'The url pointing to your cluster ingress.',
+        },
+        dockerImage: {
+          type: 'string',
+          description: 'The docker image containing taskcluster.',
+        },
+        ingressStaticIpName: {
+          type: 'string',
+          description: 'A google static ip object that the ingress can use to maintain the same ip address.',
+        },
+        ingressCertName: {
+          type: 'string',
+          description: 'A google certificate name that the ingress can use to set up tls.',
+        },
+      },
+      required: ['rootUrl', 'dockerImage'],
+      aditionalProperties: false,
+    };
+
+    // Something to copy-paste for users
+    const exampleConfig = {
+      rootUrl: '...',
+      dockerImage: '...',
+      ingressStaticIpName: '...',
+      ingressCertName: '...',
+    };
+    const variablesYAML = {}; // Defaults that people can override
+
+    let configs = SERVICES.map(name => ({
+      name,
+      vars: requirements[`configs-${name}`],
+      procs: requirements[`procslist-${name}`],
+    }));
+    configs = configs.concat(Object.entries(extras).map(([name, {vars, procs}]) => ({
+      name,
+      vars,
+      procs,
+    })));
+
+    configs.forEach(cfg => {
+      const confName = cfg.name.replace(/-/g, '_');
+      exampleConfig[confName] = {};
+      variablesYAML[confName] = {
+        procs: {},
+      };
+      schema.required.push(confName);
+      schema.properties[confName] = {
+        type: 'object',
+        title: `Configuration options for ${cfg.name}`,
+        properties: {
+          procs: {
+            type: 'object',
+            title: 'Process settings for this service',
+            properties: {},
+            required: [],
+          },
+        },
+        required: ['procs'],
+      };
+
+      // Some services actually duplicate their config env vars in multiple places
+      // so we de-dupe first. We use the variable name for this. If they've asked
+      // for the same variable twice with different types then this is not our fault
```

Comment:
Shouldn't tc-lib-config do this?

Reply:
It probably should but it was easier to do it this way for now and hopefully lib-config goes away soon.